File: static/scripts/controllers/vipController.js
myApp.controller('vipController', ['$scope', '$http',
    function($scope, $http){

        $scope.clients = {};
        $scope.clients.spent = 1000;
        $scope.clients.payments = 3;


        $scope.searchVipList = function(spent, payments){

            $scope.loading = true;

            $http({
                url: '/vip/searchVipList',
                method: "GET",
                params: {'spent': spent, 'payments': payments}
            }).then(function(response){

                $scope.vip_list = response.data;

            }).finally(function(){
                $scope.loading = false;
            });

        };


        //$scope.searchVipCount = function(spent, payments){
        //
        //    $scope.clients.vipCount = null;
        //    $scope.loading = true;
        //
        //    $http({
        //        url: '/vip/searchVipCount',
        //        method: "GET",
        //        params: {'spent': spent, 'payments': payments}
        //    }).then(function(response){
        //
        //        $scope.clients.vipCount = response.data;
        //    }).finally(function(){
        //        $scope.loading = false;
        //    });
        //
        //};


    }
]);

File: src/main/java/lodeinfo/controller/VipController.java
package lodeinfo.controller;

import lodeinfo.model.VipEntity;
import lodeinfo.repository.VipRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@RequestMapping(value = "/vip")
public class VipController {

    @Autowired
    VipRepository vipRepository;

//    @RequestMapping(
//            value = "/searchVipCount",
//            method = RequestMethod.GET
//    )
//    public int searchVipCount(@RequestParam int spent, @RequestParam int payments) {
//        return vipRepository.getVipCount(spent, payments);
//    }

    @RequestMapping(
            value = "/searchVipList",
            method = RequestMethod.GET
    )
    public List<VipEntity> searchVipList(@RequestParam int spent, @RequestParam int payments) {
        return vipRepository.getVipList(spent, payments);
    }

}
